sympy/solvers/solveset.py

```diff
@@ -77,6 +77,17 @@ def _invert_real(f, g_ys, symbol):
 
     if isinstance(f, Abs):
         g_ys = g_ys - FiniteSet(*[g_y for g_y in g_ys if g_y.is_negative])
+
+        _a = []
+        from sympy.functions.elementary.complexes import sign
+        for g_y in g_ys:
+            if g_y.is_Mul and (g_y.args[0] == -1):
+                _a.append(FiniteSet(g_y).intersect(Interval(-oo, 0)))
+            else:
+                _a.append(FiniteSet(g_y).intersect(Interval(0, oo)))
+
+        g_ys = Union(*[element for element in _a if element.args])
+
         return _invert_real(f.args[0],
                             Union(g_ys, imageset(Lambda(n, -n), g_ys)), symbol)
```

Comment:
You current approach is not quite right, see:
```
In [12]: invert_real(Abs(x), -y, x)
Out[12]: (x, Intersection([0, oo), {y}) U Intersection((-oo, 0], {-y})) # should be (x, Intersection([0, oo), {-y}) U Intersection((-oo, 0], {y})) 
In [14]: invert_real(Abs(x), -y, x)[1].subs('y', 1)
Out[14]: {-1, 1} # should be EmptySet
```

Rather what you need to do is replace the last `return` call with something like, after that line 79 will also become redundant.
```
         return _invert_real(f.args[0],
                             Union(imageset(Lambda(n, n), g_ys).intersect(Interval(0, oo)),
                                 imageset(Lambda(n, -n), g_ys).intersect(Interval(-oo, 0))),
                             symbol)
```